lib/active_admin/views/components/status_tag.rb

```diff
@@ -50,7 +47,7 @@ def build(*args)
         super(content, options)
 
         add_class(status_to_class(status)) if status
-        add_class(type.to_s) if type
+        add_class(type.to_s) if type && !type.is_a?(Hash)
```

Comment:
Regarding the implementation, after a second read I think it's better to keep the splat. I think we can actually leave this method exactly as it was, but just changing this:

```diff
- add_class(type.to_s) if type
+ if type
+   Deprecation.warn "the `type` parameter is deprecated! Provide as a class: `status_tag(status, class: 'abc')`."
+   add_class(type.to_s)
+  end
```